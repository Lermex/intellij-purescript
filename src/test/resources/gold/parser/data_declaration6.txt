Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('A')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('User6')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('User6')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(ObjectType)
            PsiElement({)('{')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(GenericIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('name')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(TypeConstructor)
                    PSASTWrapperElement(Qualified)
                      PsiElement(proper name)('String')
                      PsiElement(whitespace)(' ')
            PsiElement(})('}')