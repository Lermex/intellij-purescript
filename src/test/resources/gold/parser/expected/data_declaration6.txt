Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('A')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSDataDeclarationImpl(DATA_DECLARATION)
        PsiElement(PSTokenType.data)('data')
        PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
          PsiElement(PSTokenType.PROPER_NAME)('User6')
        PsiElement(PSTokenType.=)('=')
        PSDataDeclarationItemListImpl(DATA_DECLARATION_ITEM_LIST)
          PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
            PSDeclarationRefTypeListImpl(DECLARATION_REF_TYPE_LIST)
              PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                PsiElement(PSTokenType.PROPER_NAME)('User6')
            PSDataDeclarationBodyImpl(DATA_DECLARATION_BODY)
              PsiElement(PSTokenType.{)('{')
              PSDataDeclarationTypeDefinitionListImpl(DATA_DECLARATION_TYPE_DEFINITION_LIST)
                PSDataDeclarationTypeDefinitionImpl(DATA_DECLARATION_TYPE_DEFINITION)
                  PsiElement(PSTokenType.ID)('name')
                  PsiElement(PSTokenType.::)('::')
                  PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                    PsiElement(PSTokenType.PROPER_NAME)('String')
              PsiElement(PSTokenType.})('}')